public/app/core/components/RolePicker/TeamRolePicker.tsx

```diff
@@ -5,35 +5,51 @@ import { contextSrv } from 'app/core/core';
 import { Role, AccessControlAction } from 'app/types';
 
 import { RolePicker } from './RolePicker';
-// @ts-ignore
 import { fetchTeamRoles, updateTeamRoles } from './api';
 
 export interface Props {
   teamId: number;
   orgId?: number;
   roleOptions: Role[];
   disabled?: boolean;
-  builtinRolesDisabled?: boolean;
+  onApplyRoles?: (newRoles: Role[]) => void;
+  pendingRoles?: Role[];
+  apply?: boolean;
 }
 
-export const TeamRolePicker: FC<Props> = ({ teamId, orgId, roleOptions, disabled, builtinRolesDisabled }) => {
+export const TeamRolePicker: FC<Props> = ({
+  teamId,
+  roleOptions,
+  disabled,
+  onApplyRoles,
+  pendingRoles,
+  apply = false,
+}) => {
   const [{ loading, value: appliedRoles = [] }, getTeamRoles] = useAsyncFn(async () => {
     try {
-      return await fetchTeamRoles(teamId, orgId);
+      if (apply) {
+        if (pendingRoles?.length! > 0) {
```

Comment:
```suggestion
      if (apply && Boolean(pendingRoles?.length)) {
```
This condition can be simplified a bit.

Reply:
Done, thanks!